apps/sens/fsw/unit_test/sens_cmds_test.cpp

```diff
@@ -48,43 +51,239 @@
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SB_BAD_ARGUMENT, 2);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "CMD pipe read error (0x%08lX)", CFE_SB_BAD_ARGUMENT);
+
+    UtAssert_EventSent(SENS_RCVMSG_ERR_EID, CFE_EVS_ERROR, expectedEvent,
+                       "ProcessCmdPipe, CmdPipeError");
+}
+
+
+/**
+ * Test SENS ProcessCmdPipe, Noop
+ */
+void Test_SENS_ProcessCmdPipe_Noop(void)
 {
-    return 5;
+    SENS              oSENS;
+
+    int32             CmdPipe;
+    SENS_NoArgCmd_t   InMsg;
+    char              expectedEvent[CFE_EVS_MAX_MESSAGE_LENGTH];
+
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    CmdPipe = Ut_CFE_SB_CreatePipe("SENS_CMD_PIPE");
+    CFE_SB_InitMsg ((void*)&InMsg, SENS_CMD_MID, sizeof(InMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&InMsg, (uint16)SENS_NOOP_CC);
+    Ut_CFE_SB_AddMsgToPipe((void*)&InMsg, (CFE_SB_PipeId_t)CmdPipe);
+
+    SENS_Test_PrintCmdMsg((void*)&InMsg, sizeof(SENS_NoArgCmd_t));
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SUCCESS, 1);
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_GETMSGID_INDEX, SENS_SEND_HK_MID, 1);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "Recvd NOOP. Version %d.%d.%d.%d",
+                            SENS_MAJOR_VERSION, SENS_MINOR_VERSION,
+                            SENS_REVISION, SENS_MISSION_REV);
+
+    UtAssert_EventSent(SENS_CMD_NOOP_EID, CFE_EVS_INFORMATION, expectedEvent,
+                       "ProcessCmdPipe, Noop Event Sent");
+    UtAssert_True(oSENS.HkTlm.usCmdCnt == 1, "ProcessCmdPipe, Noop");
 }
 
 
-int32 SENS_Cmds_Test_UT_CFE_SB_SubscribeHook2(CFE_SB_MsgId_t MsgId, CFE_SB_PipeId_t PipeId,
-                                                CFE_SB_Qos_t Quality, uint16 MsgLim)
+/**
+ * Test SENS ProcessCmdPipe, Reset
+ */
+void Test_SENS_ProcessCmdPipe_Reset(void)
 {
-    return 6;
+    SENS              oSENS;
+
+    int32             CmdPipe;
+    SENS_NoArgCmd_t   InMsg;
+
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    CmdPipe = Ut_CFE_SB_CreatePipe("SENS_CMD_PIPE");
+    CFE_SB_InitMsg ((void*)&InMsg, SENS_CMD_MID, sizeof(InMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&InMsg, (uint16)SENS_RESET_CC);
+    Ut_CFE_SB_AddMsgToPipe((void*)&InMsg, (CFE_SB_PipeId_t)CmdPipe);
+
+    SENS_Test_PrintCmdMsg((void*)&InMsg, sizeof(SENS_NoArgCmd_t));
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SUCCESS, 1);
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_GETMSGID_INDEX, SENS_SEND_HK_MID, 1);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
```

Comment:
No test case.  In this case, you really should verify that the RESET command actually did a reset.  The best way to either change the Command Count to a non-zero value and verify that, after the reset, the Command Count is zero.  If the attribute is private, obviously you can't change it.  If that is the case, the best way to test it is to (in this order) send a NOOP to the command pipe, a SEND_HK to the sch pipe, a RESET to the command pipe, and finally another SEND_HK to the sch pipe.  Then, you can verify the command count in the HK message increments to 1 after the NOOP function, and goes back to 0 after the RESET function.  If the RESET command also sends an event, verify that the event is sent.

Reply:
Working on it.

Reply:
Done

Reply:
I updated code and pushed.